src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,181 @@
+  altText,
+  className,
+  titles,
+}) {
+  const baseLayers = layers.filter(layer => layer.getIsBaseLayer());
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const [isClosed, setIsClosed] = useState(true);
+  const [currentLayer, setCurrentLayer] = useState(getVisibleLayer(baseLayers));
+  /* Images are loaded from props if provided, fallback from layer */
+  const images = layerImages
+    ? Object.keys(layerImages).map(layerImage => layerImages[layerImage])
+    : baseLayers.map(layer => layer.previewImage);
+
+  if (!baseLayers || baseLayers.length < 2) {
+    return null;
+  }
+
+  useEffect(() => {
+    /* Ensure correct layer is active on app load */
+    if (currentLayer !== getVisibleLayer(baseLayers)) {
+      setCurrentLayer(getVisibleLayer(baseLayers));
+    }
+  }, [currentLayer]);
+
+  useEffect(() => {
+    /* Used for correct layer image render with animation */
+    if (!switcherOpen) {
+      const timeout = setTimeout(() => {
+        setIsClosed(true);
+      }, 200);
+      return () => clearTimeout(timeout);
+    }
+    return setIsClosed(false);
+  }, [switcherOpen]);
+
+  const toggleBtn =
+    baseLayers.length > 0 ? (
+      <div
+        className="rs-base-layer-switch-close"
+        role="button"
+        onClick={() => setSwitcherOpen(false)}
+        onKeyPress={e => e.which === 13 && setSwitcherOpen(false)}
+        tabIndex="0"
+        aria-label={titles.closeSwitcher}
+        title={titles.closeSwitcher}
+      >
+        <FaChevronCircleLeft size={15} focusable={false} />
+      </div>
+    ) : null;
+
+  return (
+    <div className={`${className}${switcherOpen ? ' open' : ''}`}>
+      <div className="rs-base-layer-switch-layers">
+        {switcherOpen ? toggleBtn : null}
+        {baseLayers.map((layer, index) => {
+          const layerName = layer.getName();
+          return (
+            <div
+              key={layer.key}
+              className={`rs-base-layer-switch-button${
+                switcherOpen ? ' open' : ''
+              }${layerName === currentLayer.getName() ? ' rs-active' : ''}`}
+              role="button"
+              title={isClosed ? titles.openSwitcher : layerName}
+              aria-label={isClosed ? titles.openSwitcher : layerName}
+              onClick={() => {
+                if (!switcherOpen) {
+                  setSwitcherOpen(true);
+                  return;
+                }
+                setCurrentLayer(layer);
+                layer.setVisible(true);
+                setSwitcherOpen(false);
+              }}
+              onKeyPress={e => {
+                if (e.which === 13) {
+                  if (!switcherOpen) {
+                    setSwitcherOpen(true);
+                  } else {
+                    layer.setVisible(true);
+                    setCurrentLayer(layer);
+                    setSwitcherOpen(false);
+                  }
+                }
+              }}
+              tabIndex={`${isClosed ? index : '0'}`}
```

Comment:
I don't think  we need different tabIndex , put 0 everywhere. it could potentially breaks the default keyboard navigation.

Reply:
If I give all of them "0" it will skip to the next button even when the switcher is closed. Ideal would be giving the first button 0 and the others -1 while closed so they are omitted when using tab.

Reply:
I wrote a function that returns the tabIndex we need

Reply:
> If I give all of them "0" it will skip to the next button even when the switcher is closed (to buttons that aren't visible).

that means after the user goes trough all the element with tabIndex 0, it will goes to element with tabIndex 1 then 2 and so on. So it just pushes the problem later. The question you must ask yourself , why the focus goes to hidden elements ?  maybe the hidden elements should not be there.

if the menu is closed displays only the button that opens the menu, if it's open, displays all the base layers button and the button to close it.

Reply:
The reason I don't just render all the buttons on switcherOpen was because I couldn't get it to work with the transition animation. Will try to find a way though

Reply:
You're right, in the meantime I can use the isClosed variable instead of switcherOpen so the animation runs smoothly on open and close